src/v1/controllers/TenantCreate/TenantController.ts

```diff
@@ -0,0 +1,55 @@
+import { Request, Response } from 'express';
+import logger from '../../utils/logger';
+import * as _ from 'lodash';
+import * as uuid from 'uuid';
+import tenantCreateJson from './createTenantValidationSchema.json';
+import { errorResponse, successResponse } from '../../utils/response';
+import httpStatus from 'http-status';
+import { createTenant, getTenant } from '../../services/tenantService';
+import { schemaValidation } from '../../services/validationService';
+
+export const apiId = 'api.tenant.create';
+
+const tenantCreate = async (req: Request, res: Response) => {
+  const requestBody = req.body;
+  const id = uuid.v4();
+  try {
+    const isRequestValid: Record<string, any> = schemaValidation(requestBody, tenantCreateJson);
+    if (!isRequestValid.isValid) {
+      const code = 'TENANT_INVALID_INPUT';
+      logger.error({ code, apiId, requestBody, message: isRequestValid.message });
+      return res.status(httpStatus.BAD_REQUEST).json(errorResponse(id, httpStatus.BAD_REQUEST, isRequestValid.message, code));
+    }
+
+    const isTenantExists = await checkTenantExists(_.get(requestBody, ['tenant_name']));
+    if (isTenantExists) {
+      const code = 'TENANT_EXISTS';
+      logger.error({ code, apiId, requestBody, message: `Tenant Already exists with name:${_.get(requestBody, ['tenant_name'])}` });
+      return res.status(httpStatus.CONFLICT).json(errorResponse(id, httpStatus.CONFLICT, `Tenant Already exists with name:${_.get(requestBody, ['tenant_name'])}`, code));
+    }
+
```

Comment:
please add comments to all code block . a one line description will help for sure.

Reply:
okay